ext/RMagick/extconf.rb

```diff
@@ -186,6 +159,27 @@ def exit_failure(msg)
       exit(1)
     end
 
+    def determine_imagemagick_package
+      exit_failure "Can't install RMagick #{RMAGICK_VERS}. Can't find pkg-config in #{ENV['PATH']}\n" unless find_executable('pkg-config')
+
+      packages = `pkg-config --list-all`.scan(/(ImageMagick\-6[\.A-Z0-9]+) .*/)
+
+      if packages.length.zero?
+        exit_failure "Can't install RMagick #{RMAGICK_VERS}. Can't find ImageMagick with pkg-config\n"
+      end
+
+      if packages.length > 1
+        msg = "\nWarning: Found more than one ImageMagick installation. This could cause problems at runtime.\n"
+        packages.each do |package|
+          msg << " - #{package[0]}\n"
+        end
+        Logging.message msg
+        message msg
+      end
+
+      packages[0][0]
```

Comment:
It's somewhat more conventional in ruby to use `.first`:

```rb
packages.first.first
# packages.first if we use `flatten` above
```

Reply:
I am not a ruby expert as you might have noticed 😄 . Changed it